docs/plugin-protocol/tfplugin5.9.proto

```diff
@@ -0,0 +1,808 @@
+// the tag of the most recent release of Terraform, and not from the main
+// branch or any other development branch.
+//
+syntax = "proto3";
+option go_package = "github.com/hashicorp/terraform/internal/tfplugin5";
+
+import "google/protobuf/timestamp.proto";
+
+package tfplugin5;
+
+// DynamicValue is an opaque encoding of terraform data, with the field name
+// indicating the encoding scheme used.
+message DynamicValue {
+    bytes msgpack = 1;
+    bytes json = 2;
+}
+
+message Diagnostic {
+    enum Severity {
+        INVALID = 0;
+        ERROR = 1;
+        WARNING = 2;
+    }
+    Severity severity = 1;
+    string summary = 2;
+    string detail = 3;
+    AttributePath attribute = 4;
+}
+
+message FunctionError {
+    string text = 1;
+    // The optional function_argument records the index position of the
+    // argument which caused the error.
+    optional int64 function_argument = 2;
+}
+
+message AttributePath {
+    message Step {
+        oneof selector {
+            // Set "attribute_name" to represent looking up an attribute
+            // in the current object value.
+            string attribute_name = 1;
+            // Set "element_key_*" to represent looking up an element in
+            // an indexable collection type.
+            string element_key_string = 2;
+            int64 element_key_int = 3;
+        }
+    }
+    repeated Step steps = 1;
+}
+
+message Stop {
+    message Request {
+    }
+    message Response {
+        string Error = 1;
+    }
+}
+
+// RawState holds the stored state for a resource to be upgraded by the
+// provider. It can be in one of two formats, the current json encoded format
+// in bytes, or the legacy flatmap format as a map of strings.
+message RawState {
+    bytes json = 1;
+    map<string, string> flatmap = 2;
+}
+
+enum StringKind {
+    PLAIN = 0;
+    MARKDOWN = 1;
+}
+
+// ResourceIdentitySchema represents the structure and types of data used to identify
+// a managed resource type. Effectively, resource identity is a versioned object
+// that can be used to compare resources, whether already managed and/or being
+// discovered.
+message ResourceIdentitySchema {
+    // IdentityAttribute represents one value of data within resource identity. These
+    // are always used in resource identity comparisons.
+    message IdentityAttribute {
+        // name is the identity attribute name
+        string name = 1;
+
+        // type is the identity attribute type
+        bytes type = 2;
+
+        // required_for_import when enabled signifies that this attribute must be
+        // defined for ImportResourceState to complete successfully
+        bool required_for_import = 3;
+
+        // optional_for_import when enabled signifies that this attribute is not
+        // required for ImportResourceState, because it can be supplied by the
+        // provider. It is still possible to supply this attribute during import
+        // to override the provider-supplied value
```

Comment:
Just a nit, since this is a provider implementation detail, it may not choose to use this value for whatever reason
```suggestion
        // provider. It is still possible to supply this attribute during import.
```